projects/igniteui-angular/src/lib/data-operations/pivot-strategy.ts

```diff
@@ -40,6 +40,16 @@ export class PivotRowDimensionsStrategy implements IPivotDimensionStrategy {
         let prevDimTopRecords = [];
         const currRows = cloneArray(rows, true);
         PivotUtil.assignLevels(currRows);
+
+        if (currRows.length === 0) {
+            for (const row of collection) {
```

Comment:
No need to loop through each record.  Also `row` here is not a dimension so there's no point in passing it in `getFieldsHierarchy` or pushing in prevRowDims.

This could be simplified to something like:

```
        if (currRows.length === 0) {
            hierarchies = PivotUtil.getFieldsHierarchy(collection, [{memberName: '', enabled: true}], PivotDimensionType.Row, pivotKeys);
            // generate flat data from the hierarchies
            data = PivotUtil.processHierarchy(hierarchies, collection ?? [], pivotKeys, 0, true);
            return data;
        }
```